File: Python/Problems/Problems/problem3285.py
import heapq
import math
import sys

from utils import print1DArray, print2DArray
from typing import List, Set
from collections import Counter

from .problem import Problem


class Problem3285(Problem):
    def test(self) -> bool:
        height = [1, 2, 3, 4, 5]
        threshold = 2
        expected = [3, 4]

        result = self.stableMountains(height, threshold)

        print1DArray(result)

        return expected == result

    def stableMountains(self, height: List[int], threshold: int) -> List[int]:
        n = len(height)
        result = []
        for i in range(n - 1):
            if height[i] > threshold:
                result.append(i + 1)

        return result

File: C++/Problems/Problems/problem_3285.cpp
#include "problem_3285.h"

bool problem_3285::test() {
    std::vector height { 1, 2, 3, 4, 5 };
    int threshold = 2;

    std::vector expected { 3, 4 };

    auto result = stableMountains(height, threshold);

    print1DVector(result);

    return result == expected;
}

std::vector<int> problem_3285::stableMountains(std::vector<int> &height, int threshold) {
    int n = height.size();
    std::vector<int> result;
    for (int i = 0; i < n - 1; ++i)
    {
        if (height[i] > threshold)
            result.push_back(i + 1);
    }

    return result;
}
